code-challenges.test.js

```diff
@@ -27,57 +27,53 @@
 
 
 
-describe('arrayShuffler', () => {
-    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-    it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
-        expect(arrayShuffler(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
-        expect(arrayShuffler(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
-    })
-  })
+// describe('arrayShuffler', () => {
+//     const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+//     const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+//     it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+//         expect(arrayShuffler(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+//         expect(arrayShuffler(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+//     })
+//   })
 
 
 // b) Create the function that makes the test pass.
 // Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
 
 // ***PSEUDOCODE***
-//create a function
+//create a function named arrayShuffler
+//parameter is array
 //remove the first element
 // use .shift method
-//math.floor/math.random
+// use math.floor/math.random method for possible solution
```

Comment:
Your code has some functionality happening on line 56 that is not reflected in your pseudocode.  If I were to try and write the code using this pseudocode, my result would look significantly different.  You can think of your pseudocode as if you are navigating someone else who is driving.  Each step and each detail is crucial.

Reply:
We are almost there, but there are still a few pieces missing in the pseudocode.  Based on your pseudocode alone, if I were to code this out here is what I came up with: 
`const arrayShuffler = (array) => {
  array.shift()
  let randomNum = Math.floor(Math.random())
  let newArray = array[randomNum]
  return newArray
}`
What is key thing is missing in the above function that you have written in your function but not your pseudocode?  Hint: take another look at line 56.  Your updated function looks great, just want to see your pseudocode match it.